zenodo/modules/spam/utils.py

```diff
@@ -26,32 +26,83 @@
+
+def send_spam_user_email(recipient, deposit=None, community=None):
     """Send email notification to blocked user after spam detection."""
     msg = Message(
-        _("Your Zenodo Upload has been automatically marked as Spam."),
+        _("Your Zenodo activity has been automatically marked as Spam."),
         sender=current_app.config.get('SUPPORT_EMAIL'),
         recipients=[recipient],
     )
-    msg.body = render_template("zenodo_spam/email/spam_user_email.tpl")
+    msg.body = render_template(
+        "zenodo_spam/email/spam_user_email.tpl",
+        community=community,
+        deposit=deposit
+        )
     send_email.delay(msg.__dict__)
 
 
-def send_spam_admin_email(deposit, user):
+def send_spam_admin_email(user, deposit=None, community=None):
     """Send email notification to admins for a spam detection."""
     msg = Message(
-        _("Zenodo Deposit Marked as Spam."),
+        _("Zenodo activity marked as spam."),
         sender=current_app.config.get('SUPPORT_EMAIL'),
         recipients=[current_app.config.get('ZENODO_ADMIN_EMAIL')],
     )
-    msg.body = render_template(
-        "zenodo_spam/email/spam_admin_email.tpl",
-        user=user,
-        deposit=deposit)
+    if deposit:
+        msg.body = render_template(
+            "zenodo_spam/email/spam_deposit_admin_email.tpl",
+            user=user,
+            deposit=deposit)
+    if community:
+        msg.body = render_template(
+            "zenodo_spam/email/spam_community_admin_email.tpl",
+            user=user,
+            community=community)
     send_email.delay(msg.__dict__)
+
+
+def spam_checking_helper(community=None, deposit=None):
+    """Checks community/deposit metadata for spam."""
+    try:
+        if current_app.config.get('ZENODO_SPAM_MODEL_LOCATION'):
+            if community:
+                task = check_metadata_for_spam.delay(
+                    community_id=community.id)
+            if deposit:
+                task = check_metadata_for_spam.delay(dep_id=str(deposit.id))
+            spam_proba = task.get(timeout=current_app.config[
+                'ZENODO_SPAM_CHECK_TIMEOUT'])
+        else:
+            spam_proba = 0
+        if spam_proba > current_app.config['ZENODO_SPAM_THRESHOLD']:
+            if not Permission(ActionNeed('admin-access')).can():
+                rs = RecordsSearch(index='records').query(
+                    Q('query_string', query="owners:{}".format(
+                        community.id_user)))
+                if not (rs.count() or Community.query.filter_by(
+                        id_user=community.id_user).count()):
+                    current_app.config['ZENODO_SPAM_HANDLING_ACTIONS'](
+                        community=community, deposit=deposit)
+    except Exception:
+        current_app.logger.exception(u'Could not check for spam')
+    if community:
+        flash("Community was successfully created.", category='success')
+
+
+def community_spam_checking_receiver(sender, community):
```

Comment:
Moving or inline-ing this function in `communities/ext.py` might be a better fit